src/backups.py

```diff
@@ -683,15 +683,15 @@ def _retrieve_s3_parameters(self) -> Tuple[Dict, List[str]]:
             )
             return {}, missing_required_parameters
 
-        # Retrieve the backup path, strip its slashes and add a "/" in the beginning of the path.
-        s3_parameters["path"] = f'/{s3_parameters["path"].strip("/")}'
-
         # Add some sensible defaults (as expected by the code) for missing optional parameters
         s3_parameters.setdefault("endpoint", "https://s3.amazonaws.com")
         s3_parameters.setdefault("region")
         s3_parameters.setdefault("path", "")
         s3_parameters.setdefault("s3-uri-style", "host")
 
+        # Retrieve the backup path, strip its slashes and add a "/" in the beginning of the path.
+        s3_parameters["path"] = f'/{s3_parameters["path"].strip("/")}'
+
```

Comment:
Maybe consider to mimic MySQL: https://github.com/canonical/mysql-k8s-operator/blob/main/lib/charms/mysql/v0/backups.py#L142-L151

We strip trailing spaces and unnecessary slashes to make sure no garbage comes from users, e.g. endlpoint="https://s3.amazonaws.com//"  or patl="alex " might cause a lot of fun with troubleshooting.

P.S. sure, good to validate/trim/strip on s3-integrator level... but since we strip path here, maybe worse to sync with MySQL approach.

Reply:
Great suggestion. I'm gonna update the PR to sync with MySQL approach.

Reply:
Updated on https://github.com/canonical/postgresql-k8s-operator/pull/166/commits/4751cbe677d5e9a39c88f0d2d431cc8c5d3f7a6c.